docker/ci/Dockerfile

```diff
@@ -1,6 +1,9 @@
-FROM rippleci/clio_clang:16
+FROM ghcr.io/xrplf/clio-gcc:12 AS clio-gcc
+FROM ghcr.io/xrplf/clio-tools:latest AS clio-tools
+
+FROM ghcr.io/xrplf/clio-clang:16
```

Comment:
nit: Maybe add `AS clio-clang` and put the three lines together, without extra space?